add-on/src/lib/ipfs-request.js

```diff
@@ -55,7 +61,9 @@ function createRequestModifier (getState, dnsLink, ipfsPathValidator) {
 
 exports.createRequestModifier = createRequestModifier
 
-function redirectToCustomGateway (requestUrl, gwUrl) {
+function redirectToIpfsTransport (requestUrl, state, featureDetector) {
+  // TODO: redirect to `ipfs://` if inBrowserWithNativeProtocol() === true
+  const gwUrl = featureDetector.embeddedNodeIsActive() ? state.pubGwURL : state.gwURL
```

Comment:
Nitpick but, I vote to keep this function simple, as it was, and perform the feature detection before it is called.